### ISSUE TITLE
Improve jsx syntax highlighting in MyST/Sphinx/Pygments

### ISSUE THREAD

**Issue by stevepiercy** [plone/volto#3038]
I'm working on fixing the `jsx` Sphinx highlighting, and the parser is super picky. This does not parse:

```jsx
<SidebarPortal selected={this.props.selected}>
  <InlineForm
    icon={<Icon size="24px" name={nameSVG} />}
    schema={schema}
    title={schema.title}
    headerActions={<button onClick={() => {}}>Action</button>}
    footer={<div>I'm footer</div>}
    onChangeField={(id, value) => {
      this.props.onChangeBlock(this.props.block, {
        ...this.props.data,
        [id]: value,
      });
    }}
    formData={this.props.data}
  />
</SidebarPortal>;
```

...but if I quote the HTML values, then it does parse:

```jsx
<SidebarPortal selected={this.props.selected}>
  <InlineForm
    icon={"<Icon size="24px" name={nameSVG} />"}
    schema={schema}
    title={schema.title}
    headerActions={"<button onClick={() => {}}>Action</button>"}
    footer={"<div>I'm footer</div>"}
    onChangeField={(id, value) => {
      this.props.onChangeBlock(this.props.block, {
        ...this.props.data,
        [id]: value,
      });
    }}
    formData={this.props.data}
  />
</SidebarPortal>;
```

My question is whether the latter form is valid syntax when actually used as code? It would be great if someone could run a quick test for me, perhaps a simple `return <h1>hello</h1>` and `return "<h1>hello</h1>"` to compare?

To go along with this, we need to make a choice about sytnax highlighting for jsx.

1. Quote HTML values
2. Leave as is, and ignore lack of syntax highlighting for many jsx code blocks
3. [Submit a PR to fix the parser](https://github.com/fcurella/jsx-lexer)

_Originally posted by @stevepiercy in https://github.com/plone/volto/issues/3035#issuecomment-1030588839_

**Comment by stevepiercy** [comment]
Here's another one. If I remove `aria-label="Cancel"`, then it gets lexed.

````
```jsx
<button className="cancel" aria-label="Cancel" onClick={() => this.onCancel()}>
  <Icon
    name={clearSVG}
    className="circled"
    size="30px"
    title={this.props.intl.formatMessage(messages.cancel)}
  />
</button>
```
````

**Comment by stevepiercy** [comment]
@tiberiuichim thank you! Fixed in #3077.

**Comment by sneridagh** [comment]
I think not... in fact I had problems in the past if used the CamelCase:

https://reactjs.org/docs/accessibility.html#wai-aria

**Comment by tiberiuichim** [comment]
@sneridagh ok, so it's not broken syntax, just the lexer being unaware of things. Ok, TIL

**Comment by stevepiercy** [comment]
A new release of jsx-lexer might have fixed this issue. I updated the lexer to lex `aria-*` attributes.

https://github.com/fcurella/jsx-lexer/releases/tag/v2.0.0

**Comment by stevepiercy** [comment]
I need to double-check for strings that contain `'`.

**Comment by stevepiercy** [comment]
I gave it a shot to check for `'` as described at the top of the issue, but I have no idea how to modify the lexer to accept it in this context.

**Comment by ujsquared** [comment]
was trying to recreate it, using a simple `myst` project with Sphinx. This is what I'm coming across when I'm using a single quote character in the syntax.  
```
/home/ujsquared/Desktop/myst-jsx-docs/jsx_test.md:3: WARNING: Lexing literal_block "import React from 'react';\n\nconst View = props => {\n  return <div>I'm the Block view component!</div>;\n};\n\nexport default View;\n" as "jsx" resulted in an error at token: "'". Retrying in relaxed mode. [misc.highlighting_failure]
```
My guess is that the lexer hasn't yet fixed it.

**Comment by stevepiercy** [comment]
@ujsquared correct, it's an issue with the upstream project, which I filed at https://github.com/pygments/pygments/issues/2816.

**Comment by ujsquared** [comment]
looks like it's only an issue with markdown files. Lexer works okay with `.rst` source files. Weird, will give it a try.

**Comment by stevepiercy** [comment]
That _is_ very weird, and an interesting observation. According to [how MyST works](https://myst-parser.readthedocs.io/en/latest/develop/background.html#the-relationship-between-myst-restructuredtext-and-sphinx), there should be no difference.

When I copy-pasted the offending code block into the Pygments demo (currently offline, see https://github.com/pygments/pygments/issues/2917) it generated an error. Alternatively, try https://pygmentify.com/ and use the JavaScript lexer to see the result.

![Image](https://github.com/user-attachments/assets/49b530df-9d60-4948-8e02-ccf8b5dd24d3)

Or this demo with react selected as the lexer, and murphy as the theme.

https://swapoff.org/chroma/playground/

![Image](https://github.com/user-attachments/assets/31a0e7a4-88bf-4dff-a3d5-8495e57d6b75)

If it helps, the highlighted token renders to HTML as:

```html
<span class="err">&#39;</span>
```
